### ISSUE TITLE
adding a image to the redirect page

### ISSUE THREAD

**Issue by C4Wiz** [jacobsalmela/pi-hole#161]
i have tried for a while now to add a logo th the black redirect page but it never shows, i put the image in the same dir as the index.html. but it just shows up empty. thext will display fine

**Comment by dschaper** [comment]
Have you modified `index.html` so that it serves up the image? Just having the image in the directory isn't enough, you need to tell `lighttpd

**Comment by C4Wiz** [comment]
i modified the index.html:

<html>
<body>
 Ad Blocked by Pi-Hole
</body>
</html>

this works fine, but when i add this:

<html>
<body>
<img src="pihole-logo.jpg" alt="pihole Logo" height="32" width="32">
 Ad Blocked by Pi-Hole
</body>
</html>

the image does not show

**Comment by dschaper** [comment]
And is the behavior the same if you access the `index.html` directly and not through a PiHole redirect?

**Comment by C4Wiz** [comment]
yes

**Comment by dschaper** [comment]
Can you access the `lighttpd` access and error logs and see if there is anything there that may be of use to help track down the source of the issue?

**Comment by C4Wiz** [comment]
where are they located?

**Comment by dschaper** [comment]
`/var/log/lighttpd/access.log` `/var/log/lighttpd/access.log`

**Comment by C4Wiz** [comment]
[Logs.zip](https://github.com/jacobsalmela/pi-hole/files/87910/Logs.zip)

**Comment by dschaper** [comment]
It looks like you're getting 200 response code for the pi-hole.jpg image. Can you zip up the image and your index.html, and your lighttpd.conf file? I'll see if I can replicate the issue on my testbed...

**Comment by C4Wiz** [comment]
[webpage.zip](https://github.com/jacobsalmela/pi-hole/files/87930/webpage.zip)

**Comment by dschaper** [comment]
Thanks, and one other question, are you getting a broken image placeholder, or just the text alone with no broken image placeholder?

**Comment by C4Wiz** [comment]
![screen shot 2016-01-12 at 1 53 08 pm](https://cloud.githubusercontent.com/assets/3883521/12276592/e00369ac-b933-11e5-8ef7-cc2c826f4a9c.png)

**Comment by dschaper** [comment]
I don't think this is a pihole issue. The page is being served correctly, it's just not seeing the image, however the log shows a response code of 200 for the image. I'm going to try to see if I get the same behavior with a plain lighttpd install and no PiHole.

**Comment by dschaper** [comment]
Well, I have to dig deeper. On a fresh Raspbian install with no PiHole the logo does display. Will keep working on it...

**Comment by C4Wiz** [comment]
thank you so much for all the help!

**Comment by jacobsalmela** [comment]
Strange...The same thing happens to me, even with different types of images.

**Comment by dschaper** [comment]
I'm kinda stumped too. Tried PNG, GIF, TIFF and all do the same. Log shows 200, and developer mode in Chrome shows no errors and acts like the image is pulled across... I'll look through `lighttpd.conf` and see if maybe something there pops out?

**Comment by jacobsalmela** [comment]
I wonder if it has something to do with the `rewrite`, but it's weird because I feel like I replaced the ads with images before, but that was a while ago.

More investigation I guess!

**Comment by dschaper** [comment]
It's something to do with the `url.rewrite`. When I comment out that line, it displays the image without any problems...

**Comment by dschaper** [comment]
Since we have the 404 catching everything, we could probably just leave out the `url.rewrite`, and perhaps the entire stanza, since everything would fall through to the 404...?

**Comment by dschaper** [comment]
Okay, so take out the `url.rewrite` and use an absolute path for the `img src` and it's looking good.

`<img src="/pihole/pihole-logo.jpg" alt="pihole Logo" height="50" width="50">`

**Comment by C4Wiz** [comment]
yep that fixed it! i have found a site that just show a message not the pihole page
![screen shot 2016-01-13 at 7 51 34 am](https://cloud.githubusercontent.com/assets/3883521/12297466/8e21cd12-b9ca-11e5-94e8-f8796114974f.png)

**Comment by C4Wiz** [comment]
i must say you guys are awsome! the support is fantastic!!

i wanted to share my redirect page:
![screen shot 2016-01-13 at 8 09 43 am](https://cloud.githubusercontent.com/assets/3883521/12298024/3ad50ba8-b9cd-11e5-94e9-4f04572c1689.png)

he is the embeded ad replacement:
![screen shot 2016-01-13 at 8 12 02 am](https://cloud.githubusercontent.com/assets/3883521/12298047/5bbed81c-b9cd-11e5-87fd-0398d4a089b5.png)

**Comment by dschaper** [comment]
Thanks for sharing! I'll take a look at that domain and see if I can figure out why it's not triggering the replacement `index.html

**Comment by dschaper** [comment]
@jacobsalmela Are you open to adjusting the lighttpd configuration? Pulling out the redirect rules shouldn't cause any problems since the 404 trap is in place. It's a quick PR... (And well tested, lol...)

**Comment by C4Wiz** [comment]
sure

**Comment by C4Wiz** [comment]
im confused, this is now closed but i didnt see a fix for the above domain issue?

**Comment by dschaper** [comment]
The `onclickads.net` issue? That is being tracked under a new issue, it's an SSL/HTTPS issue that may not be fixable. I'll submit a PR later with an updated 'lighttpd.conf' that fixes the custom images being displayed.

**Comment by dschaper** [comment]
Closing, should be resolved.

**Comment by immae1** [comment]
if someone want to have a random gif on his blockpage try this:
https://gist.github.com/immae1/f6d37a1da8c6d4ec70eebe427c4c0383
